diffkemp/diffkemp.py

```diff
@@ -26,6 +32,90 @@ def run_from_cli():
     args.func(args)
 
 
+def build(args):
+    # Generate wrapper for C/C++ compiler
+    no_rpython = args.without_rpython or shutil.which('rpython') is None
+    cc_wrapper = generate_cc_wrapper(rpython=None if no_rpython else 'rpython',
+                                     builddir=get_simpll_build_dir())
+
+    # Create temp directory and environment
+    tmpdir = mkdtemp()
+    db_filename = os.path.join(tmpdir, 'wdb')
+    environment = {
+        wrapper_env_vars['db_filename']: db_filename,
+        wrapper_env_vars['clang']: args.clang,
+        wrapper_env_vars['clang_append']: ','.join(args.clang_append)
+        if args.clang_append is not None
+        else '',
+        wrapper_env_vars['clang_drop']: ','.join(args.clang_drop)
+        if args.clang_drop is not None
+        else '',
+        wrapper_env_vars['llvm_link']: args.llvm_link,
+    }
+    environment.update(os.environ)
+
+    # Determine make args
+    make_cc_setting = 'CC="{}"'.format(cc_wrapper)
+    make_args = [args.make, '-C', args.source_dir, make_cc_setting]
+    if args.makefile is not None:
+        make_args.append('-c')
+        make_args.append(args.makefile)
+    make_target_args = make_args[:]
+    if args.target is not None:
+        for target in args.target:
+            make_target_args.append(target)
+
+    # Clean the project
+    config_log_filename = os.path.join(args.source_dir, 'config.log')
+    if os.path.exists(config_log_filename):
+        # Backup config.log
+        os.rename(config_log_filename, config_log_filename + '.bak')
+    try:
+        check_call(make_args + ['clean'], env=environment)
+    except CalledProcessError:
+        pass
+    if os.path.exists(config_log_filename + '.bak'):
+        # Restore config.log
+        os.rename(config_log_filename + '.bak', config_log_filename)
+
+    # If config.log is present, reconfigure using wrapper
+    # Note: this is done to support building with nested configure
+    if args.reconfigure and os.path.exists(config_log_filename):
+        with open(config_log_filename, 'r') as config_log:
+            configure_cmd = None
+            for line in config_log.readlines():
+                if '$' in line:
+                    configure_cmd = line.strip()
+                    break
+            if configure_cmd and make_cc_setting not in configure_cmd:
+                configure_cmd = configure_cmd.split('$ ', 1)[1]
+                configure_cmd += ' ' + make_cc_setting
+                check_call('find "{}" -name config.cache | xargs rm -f'
+                           .format(args.source_dir), shell=True)
```

Comment:
Not much, actually it would be about the same code length. Only con is Python syntax with loops/filter/map being slightly more difficult to read, but that's not enough reason to call external programs instead.

Reply:
I agree, let's use the Python way where possible (and make it as readable as we can).